File: java/05-reactor/src/test/java/com/fResult/reactor/ch5_03/FlowAndReactiveStreamsTest.java
package com.fResult.reactor.ch5_03;

import java.util.concurrent.Flow;
import org.junit.jupiter.api.Test;
import org.reactivestreams.FlowAdapters;
import org.reactivestreams.Publisher;
import reactor.adapter.JdkFlowAdapter;
import reactor.core.publisher.Flux;
import reactor.test.StepVerifier;

class FlowAndReactiveStreamsTest {
  @Test
  void convert() {
    final var original = Flux.range(0, 10);
    Flow.Publisher<Integer> rangeOfIntegersAsJdk9Flow = FlowAdapters.toFlowPublisher(original);
    Publisher<Integer> rangeOfIntegersAsReactiveStream =
        FlowAdapters.toPublisher(rangeOfIntegersAsJdk9Flow);

    StepVerifier.create(original).expectNextCount(10).verifyComplete();
    StepVerifier.create(rangeOfIntegersAsReactiveStream).expectNextCount(10).verifyComplete();

    final var rangeOfIntegersAsReactorFluxAgain =
        JdkFlowAdapter.flowPublisherToFlux(rangeOfIntegersAsJdk9Flow);
    StepVerifier.create(rangeOfIntegersAsReactorFluxAgain).expectNextCount(10).verifyComplete();
  }
}

File: kotlin/05-reactor/src/test/kotlin/com/fResult/reactor/ch5_03/FlowAndReactiveStreamsTest.kt
package com.fResult.reactor.ch5_03

import org.junit.jupiter.api.Test
import org.reactivestreams.FlowAdapters
import org.reactivestreams.Publisher
import reactor.core.publisher.Flux
import reactor.test.StepVerifier
import java.util.concurrent.Flow

class FlowAndReactiveStreamsTest {
  @Test
  fun convert(): Unit {
    val original = Flux.range(0, 10)
    val rangeOfIntegersAsJdk9Flow: Flow.Publisher<Int> = FlowAdapters.toFlowPublisher(original)
    val rangeOfIntegersAsReactiveStream: Publisher<Int> = FlowAdapters.toPublisher(rangeOfIntegersAsJdk9Flow)

    StepVerifier.create(original).expectNextCount(10).verifyComplete()
    StepVerifier.create(rangeOfIntegersAsReactiveStream).expectNextCount(10).verifyComplete()
  }
}
